app/pages/cif/project-revision/[projectRevision]/index.tsx

```diff
@@ -91,25 +96,15 @@ export function ProjectRevision({
     });
   };
 
-  const triggerFormSelfValidation = (formObject: any): [] => {
-    formObject.onSubmit({
-      preventDefault: () => {},
-      persist: () => {},
-    });
-
-    // Effectively validating the form a second time to retrieve the errors
-    const validationResult = formObject.validate(formObject.state.formData);
-    return validationResult.errors;
-  };
-
   /**
    *  Function: approve staged change, trigger an insert on the project
    *  table & redirect to the project page
    */
   const commitProject = async () => {
     const errors = [
-      ...triggerFormSelfValidation(projectFormRef.current),
-      ...triggerFormSelfValidation(projectManagerFormRef.current),
+      ...validateFormWithErrors(projectFormRef.current),
+      ...validateFormWithErrors(projectManagerFormRef.current),
```

Comment:
@AlexZorkin this is what I had in mind for refactoring, making these form also self-validating and implementing the `ISupportExternalValidation` interface